.github/workflows/docs.yml

```diff
@@ -43,6 +43,12 @@ jobs:
         run: .ci/install.sh
       - name: Script
         run: .ci/script.sh
+      - name: Deploy to GitHub Pages
+        uses: peaceiris/actions-gh-pages@v3
+        with:
```

Comment:
It looks like the Doxygen-generated HTML currently has timestamps that will change every time despite the content being the same. Consider using the `force_orphan` option to drop the old history. Alternatively, we can try to reconfigure Doxygen to avoid the timestamps, which might make the gh-pages commits actually meaningful. But there are also other things that might change underneath (e.g. Doxygen version), which is why we originally dropped the old versions.